pages/spicedb/ops/performance.mdx

```diff
@@ -0,0 +1,90 @@
+
+### How it works
+
+Each SpiceDB node maintains an in-memory cache of permissions queries it has resolved in the past. When a new permissions query is encountered by one node, its answer may be present on another node, so SpiceDB will forward the request onward to the other node to check the shared cache.
+
+For more details on how dispatching works, see the [Consistent Hash Load Balancing for gRPC] article.
+
+[consistent hash load balancing for grpc]: https://authzed.com/blog/consistent-hash-load-balancing-grpc/
+
+### Configuration in Kubernetes environments
+
+If using the [SpiceDB Operator], dispatching is enabled by default and no additional configuration is necessary.
+
+If not using it, you need to set the following flag:
+
+```sh
+--dispatch-upstream-addr=kubernetes:///spicedb.default:50053
+```
+
+where `spicedb.default` is the Kubernetes `Service` in which SpiceDB is accessible.
+
+<Callout type="info">
+  If you are deploying SpiceDB under Kubernetes, it is recommended to use the [SpiceDB Operator], which configures dispatching automatically.
+
+  [spicedb operator]: /spicedb/ops/operator
+</Callout>
+
+### Configuration in non-Kubernetes environments
+
+<Callout type="warning">
+  Non-Kubernetes based dispatching relies upon DNS updates, which means it can become stale if DNS is changing. This is not recommended unless DNS updates are rare.
+</Callout>
+
+To enable dispatch, the following flags must be specified:
+
+```sh
+spicedb serve \
+  --dispatch-cluster-enabled=true \
+  --dispatch-upstream-addr=upstream-addr \
+  ...
+```
+
+or via environment variables with the `SPICEDB_` prefix:
+
+```sh
+SPICEDB_DISPATCH_CLUSTER_ENABLED=true \
+SPICEDB_DISPATCH_UPSTREAM_ADDR=upstream-addr \
+spicedb serve ...
+```
+
+The `upstream-addr` should be the DNS address of the load balancer at which _all_ SpiceDB nodes are accessible at the default dispatch port of `:50053`.
+
+## By enabling Materialize
+
+[Materialize] is a separate service that allows for the precomputation of permission query results.
+
+If Materialize is running, SpiceDB can dispatch sub-queries to Materialize, which can significantly speed up permission checks.
+
+[Materialize]: /authzed/concepts/authzed-materialize
+
+## By enabling the schema cache
+
+The schema cache stores type definitions and caveat definitions to avoid repeatedly fetching schema information from the datastore.
+
+SpiceDB offers two caching modes:
+
+1. **Just-In-Time (JIT) Caching**: The default mode that loads definitions on-demand. Uses less memory, but it incurs a cold start penalty on first access to each definition.
+2. **Watching Cache**: An experimental mode that proactively maintains an always-up-to-date cache. This mode uses more memory but avoids cold start penalties. It is recommended when there are frequent schema changes.
```

Comment:
@josephschorr to confirm if this is accurate

Reply:
It is, but will all be going away soon-ish once the new cache lands